ovos_dinkum_listener/service.py

```diff
@@ -135,18 +136,8 @@ def __init__(self, on_ready=on_ready, on_error=on_error,
         self._before_start()  # connect to bus
         listener = self.config["listener"]
 
-        self.mic = mic or AlsaMicrophone(
-            device=listener.get("device_name") or "default",
-            sample_rate=listener.get("sample_rate", 1600),
-            sample_width=listener.get("sample_width", 2),
-            sample_channels=listener.get("sample_channels", 1),
-            chunk_size=listener.get("chunk_size", 4096),
-            period_size=listener.get("period_size", 1024),
-            multiplier=listener.get("multiplier", 1),
-            timeout=listener.get("audio_timeout", 5),
-            audio_retries=listener.get("audio_retries", 3),
-            audio_retry_delay=listener.get("audio_retry_delay", 1),
-        )
+        self.mic = mic or OVOSMicrophoneFactory.create(self.config)
```

Comment:
should we default to alsa in this config explicitly? configs in the wild will all be missing the default values for now

(and im not sure if alsa will be the default)